libs/datamodel/connectors/datamodel-connector/src/walker_ext_traits.rs

```diff
@@ -0,0 +1,124 @@
+        let field_db_names: Vec<&str> = model
+            .get_field_db_names(&self.fields().map(|f| f.field_id()).collect::<Vec<_>>())
+            .collect();
+
+        if self.is_unique() {
+            ConstraintNames::unique_index_name(model_db_name, &field_db_names, connector).into()
+        } else {
+            ConstraintNames::non_unique_index_name(model_db_name, &field_db_names, connector).into()
+        }
+    }
+}
+
+// TODO: this lifetime doesn't make sense, it's most likely wrong.
+pub trait DefaultValueExt<'db> {
+    fn constraint_name(self, connector: &dyn Connector) -> Cow<'db, str>;
+}
+
+impl<'db> DefaultValueExt<'db> for DefaultValueWalker<'_, 'db> {
+    fn constraint_name(self, connector: &dyn Connector) -> Cow<'db, str> {
+        self.mapped_name().map(Cow::from).unwrap_or_else(|| {
+            let name = ConstraintNames::default_name(
+                self.field().model().final_database_name(),
+                self.field().final_database_name(),
+                connector,
+            );
+
+            Cow::from(name)
+        })
+    }
+}
+
+pub trait RelationFieldWalkerExt<'ast> {
+    fn final_foreign_key_name(self, connector: &dyn Connector) -> Cow<'ast, str>;
+}
+
+impl<'ast> RelationFieldWalkerExt<'ast> for RelationFieldWalker<'ast, '_> {
+    fn final_foreign_key_name(self, connector: &dyn Connector) -> Cow<'ast, str> {
+        self.explicit_mapped_name().map(Cow::Borrowed).unwrap_or_else(|| {
+            let column_names: Vec<&str> = self
+                .fields()
+                .into_iter()
+                .flatten()
+                .map(|f| f.final_database_name())
+                .collect();
+            let table_name = self.model().final_database_name();
+            ConstraintNames::foreign_key_constraint_name(table_name, &column_names, connector).into()
+        })
+    }
+}
+
+pub trait PrimaryKeyWalkerExt<'ast> {
+    fn final_database_name(self, connector: &dyn Connector) -> Option<Cow<'ast, str>>;
+}
+
+impl<'ast> PrimaryKeyWalkerExt<'ast> for PrimaryKeyWalker<'ast, '_> {
+    fn final_database_name(self, connector: &dyn Connector) -> Option<Cow<'ast, str>> {
+        if !connector.supports_named_primary_keys() {
+            return None;
+        }
+
+        Some(
+            self.mapped_name().map(Cow::Borrowed).unwrap_or_else(|| {
+                ConstraintNames::primary_key_name(self.model().final_database_name(), connector).into()
+            }),
+        )
+    }
+}
+
+pub trait CompleteInlineRelationWalkerExt<'ast> {
+    /// Gives the onDelete referential action of the relation. If not defined
+    /// explicitly, returns the default value.
+    fn on_delete(self, connector: &dyn Connector, referential_integrity: ReferentialIntegrity) -> ReferentialAction;
+}
+
+impl<'ast> CompleteInlineRelationWalkerExt<'ast> for CompleteInlineRelationWalker<'ast, '_> {
+    fn on_delete(self, connector: &dyn Connector, referential_integrity: ReferentialIntegrity) -> ReferentialAction {
```

Comment:
This is now defined only in one place?

Reply:
It's better now but it's still not there, the onDelete and onUpdate in dml are optional with defaults. They should be required.